src/include/sof/ipc/topology.h

```diff
@@ -45,6 +45,12 @@ typedef uint32_t ipc_comp;
 #include <ipc4/gateway.h>
 #define ipc_from_pipe_new(x) ((struct ipc4_pipeline_create *)x)
 #define ipc_from_pipe_connect(x) ((struct ipc4_module_bind_unbind *)x)
+
+const struct comp_driver *ipc4_get_comp_drv(int module_id);
+struct comp_dev *ipc4_get_comp_dev(uint32_t comp_id);
+int ipc4_add_comp_dev(struct comp_dev *dev);
+const struct comp_driver *ipc4_get_drv(uint8_t *uuid);
+int ipc_comp_disconnect(struct ipc *ipc, ipc_pipe_comp_connect *_connect);
```

Comment:
Looks like ```ipc_comp_disconnect()``` is generic API and not a ipc4 API. This should be moved to be with the generic IPC APIs

Reply:
sure